src/test/scala/fi/oph/koski/api/OppijaUpdateSpec.scala

```diff
@@ -272,7 +272,7 @@ class OppijaUpdateSpec extends AnyFreeSpec with KoskiHttpSpec with Opiskeluoikeu
         createOpiskeluoikeus(koululainen, original, user = helsinginKaupunkiPalvelukäyttäjä)
         val opiskeluoikeus = createOpiskeluoikeus(oppija, defaultOpiskeluoikeus)
 
-        createOrUpdate(koululainen, opiskeluoikeus.copy(lähdejärjestelmänId = Some(winnovaLähdejärjestelmäId)), {
+        createOrUpdate(koululainen, opiskeluoikeus.copy(lähdejärjestelmänId = Some(winnovaLähdejärjestelmäId("win-kf040431n"))), {
```

Comment:
Testin nimi on "Estää opiskeluoikeuden siirtymisen eri henkilölle". Tästä päättelisin, että nimenomaan tarkoituksella on haluttu käyttää samaa lähdejärjestelmän id:tä, jotta varmistetaan, ettei samaa opiskeluoikeutta voi siirtää eri oppijalle.

Reply:
Ymmärtääkseni tässä tapauksessa siirtoa ei ole vielä tehty ennen createOrUpdate-kutsua ja tuo on ainoa lähdejärjestelmän ID, joka on missään vaiheessa määriteltynä.

Mielestäni tuossa testataan, että henkilö + opiskeluoikeus -rakenteen henkilöllä on eri oid kuin opiskeluoikeuden sisällä oleva oppijaOid. Tällöin lähdejärjestelmän ID:llä ei ole merkitystä.